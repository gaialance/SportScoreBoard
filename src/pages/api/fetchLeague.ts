export type ScoreBoardType = {
  homeTeam:string
  awayTeam:string
  homeTeamScore:number
  awayTeamScore:number
  date:Date
}

const defaultPremierScoreData:ScoreBoardType[] = [
  {
    homeTeam:"Nottom Forest",
    awayTeam:"Newcastle",
    homeTeamScore:2,
    awayTeamScore:4,
    date:new Date('2023-03-18 04:00:00 +08:00'),
  },
  {
    homeTeam:"Aston Villa",
    awayTeam:"Bournemouth",
    homeTeamScore:5,
    awayTeamScore:4,
    date:new Date('2023-03-18 05:00:00 +08:00'),
  },
  {
    homeTeam:"BrentFord",
    awayTeam:"Leicester",
    homeTeamScore:5,
    awayTeamScore:4,
    date:new Date('2023-03-18 10:00:00 +08:00'),
  }
]

const defaultChampionsScoreData:ScoreBoardType[] = [
  {
    homeTeam:"Nottom Forest",
    awayTeam:"Newcastle",
    homeTeamScore:2,
    awayTeamScore:4,
    date:new Date('2023-03-17 04:00:00 +08:00'),
  },
  {
    homeTeam:"Aston Villa",
    awayTeam:"Bournemouth",
    homeTeamScore:5,
    awayTeamScore:4,
    date:new Date('2023-03-17 05:00:00 +08:00'),
  },
  {
    homeTeam:"BrentFord",
    awayTeam:"Leicester",
    homeTeamScore:5,
    awayTeamScore:4,
    date:new Date('2023-03-17 10:00:00 +08:00'),
  }
]

export type League = {
  title:string
  scoreBoards?:ScoreBoardType[]
}

const defaultLeagueData:League[] = [
  {
    title:"Premier League",
    scoreBoards:defaultPremierScoreData
  },
  {
    title:"Champions League",
    scoreBoards:defaultChampionsScoreData,
  },
  {
    title:"Europa League"
  }
]



export function fetchOneLeague(
  leagueId:number
):League {
  return defaultLeagueData[leagueId]
}

export function fetchAllLeague(
):League[] {
  return defaultLeagueData
}

export function fetchAllScore(
  leagueId:number,
) {
  return defaultLeagueData[leagueId].scoreBoards
}

export function fetchOneScore(
  leagueId:number,
  scoreId:number,
) {
  if(defaultLeagueData[leagueId]?.scoreBoards){
    return defaultLeagueData[leagueId].scoreBoards?.at(scoreId)
  }
}
