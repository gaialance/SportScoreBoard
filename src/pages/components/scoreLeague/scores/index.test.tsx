import { render,screen , cleanup } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import '@testing-library/jest-dom/extend-expect'

import Score from './'
import { ScoreBoardType } from 'src/pages/api/fetchLeague'


describe("rending the scoreComponent",()=>{
    it("Create a snapshot of the render data",()=>{
        const testData:ScoreBoardType = 
            {
                homeTeam:"Nottom Forest",
                awayTeam:"Newcastle",
                homeTeamScore:2,
                awayTeamScore:4,
                date:new Date('2023-03-18 04:00:00 +08:00'),
            }

        const { container } = render( <Score {...testData}/>)

        expect(container).toMatchSnapshot()

    })

    it("start the rending the first data",()=>{
        const testData:ScoreBoardType = 
            {
                homeTeam:"Nottom Forest",
                awayTeam:"Newcastle",
                homeTeamScore:2,
                awayTeamScore:4,
                date:new Date('2023-03-18 04:00:00 +08:00'),
            }

        render( <Score {...testData}/>)


        expect( screen.getByTestId("homeTeam") ).toHaveTextContent(testData.homeTeam)
        expect( screen.getByTestId("awayTeam") ).toHaveTextContent(testData.awayTeam)
        expect( screen.getByTestId("homeTeamScore") ).toHaveTextContent(String(testData.homeTeamScore))
        expect( screen.getByTestId("awayTeamScore") ).toHaveTextContent(String(testData.awayTeamScore))
    })
})

