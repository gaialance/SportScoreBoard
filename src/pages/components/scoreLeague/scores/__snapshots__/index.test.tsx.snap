// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rending the scoreComponent Create a snapshot of the render data 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container container css-11lq3yg-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-19otjfj-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-k3f0g4-MuiGrid-root"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            data-testid="homeTeam"
          >
            Nottom Forest
          </h5>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1o7apob-MuiGrid-root"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
          >
            vs
          </h5>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-k3f0g4-MuiGrid-root"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            data-testid="awayTeam"
          >
            Newcastle
          </h5>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-k3f0g4-MuiGrid-root"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            data-testid="homeTeamScore"
          >
            2
          </h5>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1o7apob-MuiGrid-root"
        />
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-k3f0g4-MuiGrid-root"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            data-testid="awayTeamScore"
          >
            4
          </h5>
        </div>
      </div>
    </div>
  </div>
</div>
`;
